test/components/util/util.go

```diff
@@ -182,3 +192,79 @@ func filterNonControllerPods(pods *corev1.PodList) *corev1.PodList {
 	pods.Items = filteredPods
 	return pods
 }
+
+type PortForwardInfo struct {
+	ReadyChan chan struct{}
+	StopChan  chan struct{}
+	// Port forwarding only works with a pod. Even if we ask user to provide the service name, we
+	// still have to figure out the pod name and select one of the pod the service is pointing at.
+	// Instead we can rely on the fact that prometheus pods are started by statefulset and will
```

Comment:
This seems hacky, usually pod name won't be known. So it's then up to the caller to get the pod name first. Can we change this comment to reflect that? So I wouldn't mention prometheus here at all.

Reply:
I think for now it is upto the user to provide the pod name. Going forward we can add the functionality to figure out the pod name from service name.

That's what kubectl client does as well.

Reply:
I would change this commit to not mention prometheus pod.

Reply:
@surajssd comment still mentions prometheus pod.